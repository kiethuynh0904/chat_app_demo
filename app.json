{
  "name": "chat_app_demo",
  "displayName": "chat_app_demo"
}