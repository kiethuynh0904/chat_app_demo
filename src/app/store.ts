import {configureStore, ThunkAction, Action} from '@reduxjs/toolkit';
import planetRoomReducer from '../app/slices/planetRoomSlice';

export const store = configureStore({
  reducer: {
    planetRoom: planetRoomReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
