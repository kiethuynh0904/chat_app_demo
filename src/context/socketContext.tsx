import React from 'react';
import io from 'socket.io-client';
// import {API_KEY} from '@env';

const SocketContext = React.createContext(null as any);

const SocketProvider = ({children}) => {
  const ENDPOINT = 'http://10.231.31.52:3000';
  const socket = io(ENDPOINT);
  return (
    <SocketContext.Provider value={socket}>{children}</SocketContext.Provider>
  );
};

export {SocketContext, SocketProvider};
