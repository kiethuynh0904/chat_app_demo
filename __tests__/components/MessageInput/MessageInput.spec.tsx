import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import {MessageInput} from '../../../src/components/';
import {store} from '../../../src/app/store';
import {Provider} from 'react-redux';
import {SocketProvider} from '../../../src/context/socketContext';

describe('Testing message input component', () => {
  let user = {
    name: 'Test',
    id: 'test',
  };

  test('message input snapshot', () => {
    const component = render(
      <Provider store={store}>
        <SocketProvider>
          <MessageInput user={user} />
        </SocketProvider>
      </Provider>,
    );

    expect(component).toMatchSnapshot();
  });

  test('message input should be empty', () => {
    const component = render(
      <Provider store={store}>
        <SocketProvider>
          <MessageInput user={user} />
        </SocketProvider>
      </Provider>,
    );
    const messageInp = component.getByPlaceholderText('Aa') as any;
    expect(messageInp.props.value).toBe('');
  });

  test('message input should be changed', () => {
    const component = render(
      <Provider store={store}>
        <SocketProvider>
          <MessageInput user={user} />
        </SocketProvider>
      </Provider>,
    );
    const messageInp = component.getByPlaceholderText('Aa') as any;
    fireEvent.changeText(messageInp, 'hello world');
    expect(messageInp.props.value).toBe('hello world');
  });

  test('message input should be clear text when submit', () => {
    const component = render(
      <Provider store={store}>
        <SocketProvider>
          <MessageInput user={user} />
        </SocketProvider>
      </Provider>,
    );

    const sendMsgBtn = component.getByTestId('sendMessageBtn');
    const messageInp = component.getByPlaceholderText('Aa') as any;

    fireEvent.changeText(messageInp, 'hello world');
    fireEvent.press(sendMsgBtn);

    expect(messageInp.props.value).toBe('');
  });
});
