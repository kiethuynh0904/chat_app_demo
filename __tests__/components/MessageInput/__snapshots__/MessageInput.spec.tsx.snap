// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing earth screen MessageInput snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f2f2f2",
        "borderColor": "#dedede",
        "borderRadius": 25,
        "borderWidth": 1,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginRight": 10,
        "padding": 8,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 20,
          },
          Object {
            "marginHorizontal": 5,
          },
          Object {
            "fontFamily": "simple-line-icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <TextInput
      clearButtonMode="always"
      onChangeText={[Function]}
      placeholder="Aa"
      style={
        Object {
          "flex": 1,
        }
      }
      value=""
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#3777f0",
        "borderRadius": 25,
        "height": 45,
        "justifyContent": "center",
        "width": 45,
      }
    }
    testID="sendMessageBtn"
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 18,
          },
          undefined,
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`Testing earth screen MessageInput without subscribe 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f2f2f2",
        "borderColor": "#dedede",
        "borderRadius": 25,
        "borderWidth": 1,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginRight": 10,
        "padding": 8,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 20,
          },
          Object {
            "marginHorizontal": 5,
          },
          Object {
            "fontFamily": "simple-line-icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <TextInput
      clearButtonMode="always"
      onChangeText={[Function]}
      placeholder="Aa"
      style={
        Object {
          "flex": 1,
        }
      }
      value=""
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#3777f0",
        "borderRadius": 25,
        "height": 45,
        "justifyContent": "center",
        "width": 45,
      }
    }
    testID="sendMessageBtn"
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 18,
          },
          undefined,
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`Testing earth screen message input snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f2f2f2",
        "borderColor": "#dedede",
        "borderRadius": 25,
        "borderWidth": 1,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginRight": 10,
        "padding": 8,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 20,
          },
          Object {
            "marginHorizontal": 5,
          },
          Object {
            "fontFamily": "simple-line-icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <TextInput
      clearButtonMode="always"
      onChangeText={[Function]}
      placeholder="Aa"
      style={
        Object {
          "flex": 1,
        }
      }
      value=""
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#3777f0",
        "borderRadius": 25,
        "height": 45,
        "justifyContent": "center",
        "width": 45,
      }
    }
    testID="sendMessageBtn"
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 18,
          },
          undefined,
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`Testing message input component message input snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f2f2f2",
        "borderColor": "#dedede",
        "borderRadius": 25,
        "borderWidth": 1,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginRight": 10,
        "padding": 8,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 20,
          },
          Object {
            "marginHorizontal": 5,
          },
          Object {
            "fontFamily": "simple-line-icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <TextInput
      clearButtonMode="always"
      onChangeText={[Function]}
      placeholder="Aa"
      style={
        Object {
          "flex": 1,
        }
      }
      value=""
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#3777f0",
        "borderRadius": 25,
        "height": 45,
        "justifyContent": "center",
        "width": 45,
      }
    }
    testID="sendMessageBtn"
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 18,
          },
          undefined,
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
